### Materials

- A pair of Apple Earbuds
- Ruler
- Stopwatch
- Camera
- Something to hang the pendulum off of (a desk for me)

### Procedures

1. Set up the pendulum
1. Measure the length of the pedulum
2. Pull the pendulum back and measure the angle
3. Start the timer and simultaneously let go of the pendulum
4. Record the time when the pendulum swings from the starting position to the other side and back again
5. Average the recorded times
5. Repeat the process but change the angle
6. Use the formula 'T = 2π√(l/g)' and rearrange for g
7. Calculate the percentage error of the experimental equation and the actual value of g (9.8 m/s^2)


In [25]:
import math

t1 = [2.35, 4.75-2.35, 7.09-4.75, 9.46-7.09, 11.83-9.46, 14.22-11.83] #90 degree angle
t2 = [2.08, 4.15-2.08, 6.22-4.15, 8.33-6.22, 10.42-8.33, 12.52-10.42] #45 degree angle
t3 = [2.05, 4.03-2.05, 6.07-4.03, 8.11-6.07, 10.14-8.11, 12.18-10.14] #25 degree angle
t4 = [1.83, 4.10-1.83, 6.12-4.10, 8.02-6.12, 9.99-8.02, 12.12-9.99] #2 degree angle

L = 1 #meter

#Average time of t1
avgT1 = 0
avgT2 = 0 
avgT3 = 0
avgT4 = 0
for i in t1:
    avgT1 += i
avgT1 = avgT1/len(t1)


for i in t2:
    avgT2 += i
avgT2 = avgT2/len(t2)


for i in t3:
    avgT3 += i
avgT3 = avgT3/len(t3)

for i in t4:
    avgT4 += i
avgT4 = avgT4/len(t4)

#Percent Error
real_g = 9.8

# Formula
g = L/pow(avgT1/(2*math.pi),2)
percent_err = abs((g - real_g)/ real_g) * 100
print("average period for dataset 1: {} seconds".format(avgT))
print("percent error for dataset 1: {}%".format(percent_err))
print("estimated gravity for dataset 1: {} m/s^2\n".format(g))

g = L/(pow(avgT2/(2*math.pi),2))
percent_err = abs((g - real_g)/ real_g) * 100
print("average period for dataset 2: {} seconds".format(avgT2))
print("percent error for dataset 2: {}%".format(percent_err))
print("estimated gravity for dataset 2: {} m/s^2\n".format(g))

g = L/(pow(avgT3/(2*math.pi),2))
percent_err = abs((g - real_g)/ real_g) * 100
print("average period for dataset 3: {} seconds".format(avgT2))
print("percent error for dataset 3: {}%".format(percent_err))
print("estimated gravity for dataset 3: {} m/s^2\n".format(g))

g = L/(pow(avgT4/(2*math.pi),2))
percent_err = abs((g - real_g)/ real_g) * 100
print("average period for dataset 4: {} seconds".format(avgT4))
print("percent error for dataset 4: {}%".format(percent_err))
print("estimated gravity for dataset 4: {} m/s^2".format(g))







average period for dataset 1: 0.48499999999999993 seconds
percent error for dataset 1: 28.28054692751681%
estimated gravity for dataset 1: 7.028506401103353 m/s^2

average period for dataset 2: 2.0866666666666664 seconds
percent error for dataset 2: 7.481729841442722%
estimated gravity for dataset 2: 9.066790475538614 m/s^2

average period for dataset 3: 2.0866666666666664 seconds
percent error for dataset 3: 2.244413607990731%
estimated gravity for dataset 3: 9.580047466416909 m/s^2

average period for dataset 4: 2.02 seconds
percent error for dataset 4: 1.2741407796218893%
estimated gravity for dataset 4: 9.675134203597056 m/s^2


### Conclusion

#### Claim: 
The closest average acceleration of gravity is __9.675134203597056 m/s^2__ which was when the angle was the smallest at 2º or 0.0349066rad. However when the angle was increased to 90º or π/2 rad the period formula began to break down and gave answer that had a ≈28% error rate. Therefore the pendulum period formula does not provide good support for the gravitational acceleration

#### Reasoning:
The percent error of pendulum was off by ≈28% at 90º which is the worst case scenario when it comes to justifying the gravitional effect on the pendulum. This is due to the fact that the angle in which the pendulum is dropped is not taken into consideration when doing the calculations. With the period of oscillation being modeled by a sine curve, with an increasing angle the oscillation time spikes exponentially past a certain point. This point is known as the small angle approximation and is __Θ << 1rad__ which is about .1rad or 6º according to Wikipedia. 

In order to incorporate the angle into the equation for an actual pendulum, I would need to use the differential equation:
```d^2Θ/dt^2 + (g/l)*sinΘ = 0```
Which is derived from the force equation `F = ma`, `a = -g*sinΘ` ,and `s = lΘ` which is linear acceleration. Then take the double derivative of linear acceleration and set that equal to `a = -g*sinΘ` which yields the differential equation ```d^2Θ/dt^2 + (g/l)*sinΘ = 0```

#### Sources
- https://www.math24.net/nonlinear-pendulum (in this case they solved using inertia)
- https://en.wikipedia.org/wiki/Pendulum_(mathematics)
- https://en.wikipedia.org/wiki/Pendulum
- https://www.acs.psu.edu/drussell/Demos/Pendulum/Pendula.html



